kyo-prelude/shared/src/main/scala/kyo2/kernel/TracePool.scala

```diff
@@ -0,0 +1,99 @@
+package kyo2.kernel
+
+import internal.*
+import java.util.Arrays
+import java.util.concurrent.atomic.AtomicInteger
+import scala.annotation.tailrec
+
+private[kernel] object TracePool:
+    inline def globalCapacity: Int = 1024
+    inline def localCapacity: Int  = 32
```

Comment:
These defaults should work well for most use cases but it could introduce a significant increase in memory footprint for systems that rely on blocking of large numbers of threads with Loom. We're not advertising the Loom integration yet but we might need to provide an option to disable the local caches if it becomes a first-class feature.